accepted/2025/cli-acquisition-tool.md

```diff
@@ -221,3 +236,111 @@ _Where_ the SDK is resolved from is largely orthogonal to _how compatible_ that
 The SDK team plans to release an initial version of `dnup` covering environment setup and basic install functionality as soon as possible. We then want to gather feedback from users and iterate on the feature set until the tool is a consistent, one-stop-shop for acquiring and managing .NET tooling of all kinds.
 
 The end goal is to stabilize the core workflows and experiences enough to justify the tool being as assumed part of the `dotnet` CLI's tooling capabilities - we want the SDK to be able to rely on `dnup` and use it to orchestrate more complex workspace-wide gestures like a toolset-level `dotnet restore` kind of operation that gets all of the SDKs and Runtimes needed for a workspace.
+
+## Milestones
+
+This is a large effort, and there are different areas of the work that will progress at different speeds. In addition, for compliance reasons we need to ensure that the tool is secure and robust before we can recommend it for broad usage. As a result, we expect to have multiple milestones along the way. Each of these milestones will likely be usable by motivated users, be we will not recommend them for broad usage until we reach the Public Preview milestone.
+
+* **MVP**
+  * at this phase we will be able to do the core actions, like installing publicly-released SDKs
+* **Internal Preview**
+  * at this phase we'll be able to do more lifecycle management, onboarding, checking for updates
+  * we may be missing crypto validation or other security features that would prevent a general public preview
+  * this is the phase where we should receive core end user feedback and iterate on the main UX
+* **Public Preview**
+  * at this phase we'll have all of the security related requirements and will be ready for broad usage
+* **General Availability**
+  * fit-and-finish work, documentation, telemetry, etc will all be present before we reach this milestone
+
+More details on these proposed milestones will likely vary, but may look like:
+
+### MVP
+
+* can download a platform-specific `dnup` binary from a GitHub release or other central location
+* can use `dnup` to install a specific public version of the .NET SDK to a (configurable) user-local location
+* can immediately use that user-local install
+
+### Internal Preview
+
+* can also use `global.json` information to derive the version(s) of the SDK to install
+* can check for updates to installed SDKs
+* installs are tracked by `dnup` for future management scenarios
+
+### Public Preview
+
+* settle on the name for the tool
+* uninstall of installed SDK
+* signature validation of manifests and downloaded artifacts
+* interactive UX, prompting, progress
+
+### GA
+
+* self-update of the `dnup` binary is implement
+* telemetry is implemented and documented
+* public documentation is created
+* public download url/script/mechanism is up
+* manual update check command is implemented
+
+## Other Concerns
+
+These milestones are _people_-focused. We have other use cases:
+
+### Aspire
```

Comment:
I think it would be good to call out the AOT requirement they have.

Reply:
This is also our requirement though - we wouldn't ship anything that wasn't AOT compatible.